MaxText/maxtext_utils.py

```diff
@@ -92,11 +92,24 @@ def get_train_input_output_trees(func, input_args, input_kwargs):
 # https://arxiv.org/pdf/2204.02311.pdf Appendix B
 def calculate_tflops_training_per_device(num_model_parameters, config, log=True):
   """ Calculate training TFLOP"""
-  learnable_weight_tflops = 6 * num_model_parameters * config.max_target_length * config.per_device_batch_size \
+  if config.use_megatron_tflops:
+    attention_flops = 2 * config.per_device_batch_size * config.max_target_length**2 * config.num_query_heads * config.head_dim
+    ffn1_flops = 2 * config.per_device_batch_size * config.max_target_length * config.mlp_dim * config.emb_dim * len(config.mlp_activations)
+    ffn2_flops = 2 * config.per_device_batch_size * config.max_target_length * config.mlp_dim * config.emb_dim
+    qkv_flops = 6 * config.per_device_batch_size * config.max_target_length * config.emb_dim * config.num_query_heads * config.head_dim
```

Comment:
Watch out! This should be
```
2 * config.per_device_batch_size * config.max_target_length * config.emb_dim * (config.num_query_heads + 2 * config.num_kv_heads) * config.head_dim
```

I don't think this resolves our gap because llama7b is MHA